### PR TITLE
First version of driver bms

### PR DESCRIPTION
**Pull request by RobertoAldea**

### CODE REVIEW

File: code/drv_bms/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+python_can>=4.2.2
```
Comment (mariuscrsn) [inline code/drv_bms/requirements.txt]: Can is not mandatory

File: code/drv_bms/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+system-config-tool>=0.0.6
```
Comment (mariuscrsn) [inline code/drv_bms/requirements.txt]: Same

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+from system_shared_tool import SysShdIpcChanC # pylint: disable=wrong-import-position
+
+#######################          PROJECT IMPORTS         #######################
+from can_sniffer import DrvCanCmdTypeE, DrvCanCmdDataC, DrvCanFilterC, DrvCanMessageC
+from wattrex_driver_base import DrvBaseStatusE, DrvBaseStatusC
+
+#######################          MODULE IMPORTS          #######################
+
+#######################              ENUMS               #######################
+_MAX_MSG = 100
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py:28]: Move to any common file in order to centralize data used to manage time, and chans names and size

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    ERROR_TEMP = 0x10
+    ERROR_PRES = 0x20
+
+#######################             CLASSES              #######################
+class DrvBmsDataC():
+    '''
+    Class method to create a DRVDB model of database BMS.
+    '''
+    def __init__(self, list_measures: list) -> None:
+        log.info(f"Writting data: {list_measures}")
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Change log msg

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+        Args:
+            new_status (DrvBaseStatusC): [description]
+        """
+        if isinstance(new_status, DrvBaseStatusC):
+            self.__status = new_status
+        elif isinstance(new_status, DrvBaseStatusE):
+            self.__status = DrvBaseStatusC(new_status)
+
+#     def __str__(self) -> str:
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Implement print function

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+                    else:
+                        # Iterate self.__raw_data every 2 elements and combine them into one int
+                        new_data = []
+                        for i in range(0, len(self.__raw_data), 2):
+                            new_elem = self.__raw_data[i+1] << 8 | self.__raw_data[i]
+                            new_data.append(new_elem)
+
+                        self.__data = DrvBmsDataC(new_data)
+                        # add __str__ function
+                        log.info(f"Data received from BMS: {self.__data.__dict__}")
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Change it when str function is implemented

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py:1]: Detect non reception timeout and add COMM ERROR on status

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+# TODO: improve comment and add info of bms used
+
+
+class DrvBmsDeviceC:
+    "Principal class of BMS"
+    def __init__(self, dev_id: int, can_id: int, rx_chan_name: str = 'RX_CAN_BMS') -> None:
+
+        self.dev_id: dev_id
+        self.can_id: int = (int(0x100) | can_id) & 0x7FF
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: change to private attri: `__can_id`

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+#         Raises:
+#             - None.
+#         '''
+
+# TODO: improve comment and add info of bms used
+
+
+class DrvBmsDeviceC:
+    "Principal class of BMS"
+    def __init__(self, dev_id: int, can_id: int, rx_chan_name: str = 'RX_CAN_BMS') -> None:
```
Comment (mariuscrsn) [inline code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: remove can_id

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py (hunk 8 above)
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Can id will always be same as dev id??

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py (hunk 7 above)
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Done

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py (hunk 6 above)
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Done, check if str function is enough or should include something more

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py (hunk 5 above)
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Done

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py:1]: Done

File: code/drv_bms/src/wattrex_driver_bms/drv_bms.py (hunk 8 above)
Comment (RobertoAldea) [reply code/drv_bms/src/wattrex_driver_bms/drv_bms.py]: Omit review

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Pull request merged**